test/unittests/t_sync_union_tarball.cc

```diff
@@ -0,0 +1,174 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <sync_union_tarball.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <util/string.h>
+
+#include <fstream>
+#include <iostream>
+
+#include "mock/m_sync_mediator.h"
+
+#include "sync_item.h"
+
+#include "tar_files.h"
+
+using ::testing::_;
+using ::testing::DefaultValue;
+using ::testing::Return;
+using ::testing::Property;
+
+using ::publish::SyncItem;
+
+namespace {
+
+class T_Sync_union_tarball : public ::testing::Test {
+ protected:
+  void SetUp() {
+    zlib::Algorithms algo;
+    DefaultValue<zlib::Algorithms>::Set(algo);
+
+    m_sync_mediator = new publish::MockSyncMediator();
+  }
+
+  std::string CreateTarFile(std::string tar_filename, std::string base64_data) {
+    std::ofstream tar;
+    std::string data_binary;
+    Debase64(base64_data, &data_binary);
+
+    tmp_tar_filename = "/tmp/" + tar_filename;
+
+    tar.open(tmp_tar_filename.c_str(), std::ios::binary);
+
+    if (!tar.is_open()) {
+      cout << "Error: " << strerror(errno);
+      assert(tar.is_open());
+    }
+
+    assert(tar.write(data_binary.c_str(), data_binary.size()));
```

Comment:
You should find the necessary convenience routines in util/posix that allow you to not use streams.